### ISSUE TITLE
no-missing-end-of-source-newline false positive after upgrade to v16.4.0 (from 15.11.0)

### ISSUE THREAD

**Issue by TuuZzee** [stylelint/stylelint#7669]
### What minimal example or steps are needed to reproduce the bug?

```css
  .bp5-progress-bar.bp5-intent-success .bp5-progress-meter {
    background-color: #0f9960;
  }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "extends": ["stylelint-config-standard", "stylelint-config-recommended", "stylelint-config-standard-scss", "stylelint-config-styled-components"],
  "customSyntax": "postcss-styled-syntax",
  "rules": {
    "declaration-empty-line-before": null,
    "function-calc-no-unspaced-operator": null,
    "scss/operator-no-unspaced": null,
    "scss/operator-no-newline-after": null,
    "keyframes-name-pattern": null,
    "no-descending-specificity": null,
    "selector-type-case": ["lower", { "ignoreTypes": ["/^\\$\\w+/"] }],
    "selector-type-no-unknown": [true, { "ignoreTypes": ["/-styled-mixin/", "/^\\$\\w+/"] }],
    "value-keyword-case": ["lower", { "ignoreKeywords": ["dummyValue"] }]
  }
}
```

### How did you run Stylelint?

```sh
stylelint './**/*.{ts,tsx,css}' --config .stylelintrc --max-warnings=0
```

### Which Stylelint-related dependencies are you using?

```json
    "stylelint": "16.4.0",
    "stylelint-config-recommended": "14.0.0",
    "stylelint-config-standard-scss": "13.1.0",
    "stylelint-config-styled-components": "0.1.1",
    "stylelint-processor-styled-components": "1.10.0",
```

### What did you expect to happen?

No false positive on stylelint run, it seems that rule is suppose to be deprecated since version [15](https://stylelint.io/changelog/). I have also tried to disable the rule manually (`"no-missing-end-of-source-newline": null,`) without any success

### What actually happened?

The error output on a lot of files as such: `✖  Unknown rule no-missing-end-of-source-newline  no-missing-end-of-source-newline`

### Do you have a proposal to fix the bug?

No proposal to provide, just a thank you for this linter tool, it's pretty neat!

**Comment by Mouvedia** [comment]
The problem comes from here: https://github.com/styled-components/stylelint-config-styled-components/blob/41f206e288e1e0b6907a717bbec5d6b3464db2b2/index.js#L6
`stylelint-config-styled-components` last update was 5 years ago, which is before [the removal of `processors`](https://stylelint.io/migration-guide/to-15/#removed-processors-configuration-property).

You should probably use `"customSyntax": "postcss-styled-syntax"` instead.
ref https://styled-components.com/docs/tooling#stylelint
